crates/ef-testing/src/evm_sequencer/account/v1.rs

```diff
@@ -66,25 +38,70 @@ impl KakarotAccount {
         let mut storage = vec![
             starknet_storage!(ACCOUNT_EVM_ADDRESS, evm_address),
             starknet_storage!(ACCOUNT_IS_INITIALIZED, 1u8),
+            starknet_storage!(ACCOUNT_BYTECODE_LEN, code.len() as u32),
         ];
 
         // Write the nonce of the account is written to storage after each tx.
         storage.append(&mut vec![starknet_storage!(ACCOUNT_NONCE, nonce)]);
 
-        // Initialize the bytecode storage vars.
-        // Assumes that the bytecode is stored as a ByteArray type, following the Store<ByteArray> implementation of
-        // the cairo core library
-        let mut bytecode_storage = prepare_bytearray_storage(code);
+        // Initialize the bytecode storage var.
+        let mut bytecode_storage = pack_byte_array_to_starkfelt_array(code)
+            .enumerate()
+            .map(|(i, bytes)| (StorageKey::from(i as u32), bytes))
+            .collect();
         storage.append(&mut bytecode_storage);
 
+        // Initialize the code hash var
+        let account_is_empty =
+            code.is_empty() && nonce == Felt::from(0) && balance == U256::from(0);
+        let code_hash = if account_is_empty {
+            U256::from(0)
```

Comment:
Don't think this is ever valid, empty account should be keccak empty too I believe

Reply:
no, empty (non-existing) accounts codehash are zero, not empty keccak

Reply:
Can you send spec? Yellow paper and reth use keccak empty

Reply:
https://eips.ethereum.org/EIPS/eip-1052

Reply:
Ok this is a problem for the rpc side I think, because extcode hash needs to return 0 but I don't think we should return 0 on the rpc (cfr screenshot)

![image](https://github.com/user-attachments/assets/ea4a7ee9-12e8-4fdd-8636-a5aac4d4aeb4)

Reply:
@greged93 It is possible that the yellow paper is not correctly updated after each merged EIP, it's a bit sad but it is not a reliable source of info when it comes to very specific details :/

Reply:
No no the yellow paper is correct: the code hash is empty keccak. But the extcode hash returns 0 on dead contracts. It's just not the same afaiu

Reply:
ah, ok; but regardless it has happened multiple times that the yellow paper was not exactly matching execution-specs

Reply:
I'll make a query to a geth node to check what it does just to be sure

Reply:
when in doubt -> https://github.com/ethereum/execution-specs/blob/07f5747a43d62ef7f203d41d77005cb15ca5e434/src/ethereum/cancun/vm/instructions/environment.py#L477-L481